extensions/smallrye-health/runtime/src/main/java/io/quarkus/smallrye/health/runtime/QuarkusAsyncHealthCheckFactory.java

```diff
@@ -27,7 +29,13 @@ public QuarkusAsyncHealthCheckFactory(Vertx vertx) {
     @Override
     public Uni<HealthCheckResponse> callSync(HealthCheck healthCheck) {
         Uni<HealthCheckResponse> healthCheckResponseUni = super.callSync(healthCheck);
-        return healthCheckResponseUni.runSubscriptionOn(MutinyHelper.blockingExecutor(vertx, false));
+        return healthCheckResponseUni.runSubscriptionOn(command -> {
+            Context duplicatedContext = VertxContext.createNewDuplicatedContext(vertx.getOrCreateContext());
+            duplicatedContext.executeBlocking(() -> {
```

Comment:
Let's avoid the lambda, and use an anonymous class instead.

Reply:
OK, but can you explain why? Performance?

Reply:
Yes, performances.